contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -0,0 +1,681 @@
+    /// @dev This mapping is intended to mirror the _coverLimitOf mapping, except time between cooldown start and after cooldown completed
+    mapping (uint256 => uint256) internal _preDeactivateCoverLimitOf;
+
+    /// @notice The policy holder => reward points. Having a reward points mechanism enables `free` cover gifts and discounts for referrals.
+    mapping (address => uint256) internal _rewardPointsOf;
+
+    /// @notice PolicyID => Has referral code been used for this policyID?
+    mapping (uint256 => bool) internal _isReferralCodeUsed;
+
+    /***************************************
+    MODIFIERS
+    ***************************************/
+
+    modifier whileUnpaused() {
+        require(!_paused, "contract paused");
+        _;
+    }
+
+    /**
+     * @notice Constructs `Soteria` product.
+     * @param governance_ The governor.
+     * @param registry_ The [`Registry`](./Registry) contract address.
+     * @param domain_ The user readable name of the EIP712 signing domain.
+     * @param version_ The current major version of the signing domain.
+     */
+    constructor(
+        address governance_,
+        address registry_,
+        string memory domain_,
+        string memory version_
+    ) ERC721("Soteria Policy", "SOPT") Governable(governance_) EIP712(domain_, version_) {
+        require(registry_ != address(0x0), "zero address registry");
+        _registry = IRegistry(registry_);
+        require(_registry.get("riskManager") != address(0x0), "zero address riskmanager");
+    }
+    
+    /***************************************
+    FALLBACK FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Fallback function to allow contract to receive *ETH*.
+    */
+    // solhint-disable-next-line 
+    receive() external payable nonReentrant {
+        _deposit(msg.sender, msg.value);
+    }
+
+    /**
+     * @notice Fallback function to allow contract to receive *ETH*.
+    */
+    fallback() external payable nonReentrant {
+        _deposit(msg.sender, msg.value);
+    }
+
+    /***************************************
+    POLICY HOLDER FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Activates policy on the behalf of the policyholder.
+     * @param policyholder_ Holder of the position to cover.
+     * @param coverLimit_ The value to cover in **ETH**.
+     * @param referralCode_ Referral code
+     * @return policyID The ID of newly created policy.
+    */
+    function activatePolicy(address policyholder_, uint256 coverLimit_, uint256 referralCode_) external payable override whileUnpaused returns (uint256 policyID) {
+        require(policyholder_ != address(0x0), "zero address policyholder");
+        require(coverLimit_ > 0, "zero cover value");
+        
+        policyID = policyOf(policyholder_);
```

Comment:
What if user doesn’t have a policyID yet? I assume policyID would be zero then?

Reply:
It means this is the new user and we should mint soteria policy for him. Other than, it means user has a soteria policy(activated/deactiveted) and we don't mint policy for him.

Reply:
I understand that, the question is: what does `policyOf(policyholder_)` returns if no object is stored in the mapping?

Reply:
Returns 0. The policy ids starts from 1.